BraveWallet/Crypto/Transaction Confirmations/TransactionConfirmationView.swift

```diff
@@ -21,21 +24,92 @@ struct TransactionConfirmationView: View {
   }
   
   @State private var viewMode: ViewMode = .transaction
-  @State private var activeTransactionIndex: Int = 0
+  @State private var activeTransactionId: BraveWallet.TransactionInfo.ID = "" {
+    didSet {
+      confirmationStore.fetchDetails(for: activeTransaction)
+    }
+  }
   
   private func next() {
-    activeTransactionIndex += 1
-    if activeTransactionIndex == transactions.endIndex {
-      activeTransactionIndex = 0
+    if let index = transactions.firstIndex(where: { $0.id == activeTransactionId }) {
+      var nextIndex = transactions.index(after: index)
+      if nextIndex == transactions.endIndex {
+        nextIndex = 0
+      }
+      activeTransactionId = transactions[nextIndex].id
+    } else {
+      activeTransactionId = transactions.first!.id
```

Comment:
can this be empty at any point?

Reply:
No this screen can never be presented unless transactions is non-empty. It is also guarded against this here: https://github.com/brave/brave-ios/pull/4567/files#diff-b5e46895fcabbaf5ddbc0680da3204666fcf76f65883af6f7a4e3e265a5ead3cR97